src/System.Linq.Expressions/tests/Interpreter/InterpreterTests.Call.cs

```diff
@@ -0,0 +1,137 @@
+
+#if FEATURE_INTERPRET
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using Xunit;
+
+namespace Tests.Expressions
+{
+    partial class InterpreterTests
+    {
+        [Fact(Skip = "4150")]
+        public static void CompileInterpretCrossCheck_Call_WriteBacks()
+        {
+            foreach (var e in Call_WriteBacks())
+            {
+                Verify(e);
+            }
+        }
+
+        private static IEnumerable<Expression> Call_WriteBacks()
+        {
+            // OK
+            foreach (var t in new[] { typeof(C), typeof(S) })
+            {
+                var p = Expression.Parameter(t);
+                var m = Expression.PropertyOrField(p, "X");
+                var mtd = t.GetTypeInfo().GetDeclaredMethod("Do");
+
+                yield return Expression.Block(new[] { p }, Expression.Assign(p, Expression.New(t)), Expression.Call(p, mtd), m);
+            }
+
+            // OK
+            foreach (var t in new[] { typeof(C), typeof(S) })
+            {
+                var p = Expression.Parameter(t.MakeArrayType());
+                var o = Expression.ArrayIndex(p, Expression.Constant(0));
+                var m = Expression.PropertyOrField(o, "X");
+                var mtd = t.GetTypeInfo().GetDeclaredMethod("Do");
+
+                yield return Expression.Block(new[] { p }, Expression.Assign(p, Expression.NewArrayInit(t, Expression.New(t))), Expression.Call(o, mtd), m);
+            }
+
+            // OK
+            foreach (var t in new[] { typeof(C), typeof(S) })
+            {
+                var p = Expression.Parameter(t.MakeArrayType(2));
+                var o = Expression.ArrayAccess(p, Expression.Constant(0), Expression.Constant(0));
+                var m = Expression.PropertyOrField(o, "X");
+                var mtd = t.GetTypeInfo().GetDeclaredMethod("Do");
+
+                yield return Expression.Block(new[] { p }, Expression.Assign(p, Expression.NewArrayBounds(t, Expression.Constant(1), Expression.Constant(1))), Expression.Assign(o, Expression.New(t)), Expression.Call(o, mtd), m);
+            }
+
+            // FAIL - Issue for struct; compiler doesn't write back but interpreter does
+            foreach (var t in new[] { typeof(C), typeof(S) })
+            {
+                var p = Expression.Parameter(typeof(Holder<>).MakeGenericType(t));
+                var o = Expression.Property(p, "Value");
+                var m = Expression.PropertyOrField(o, "X");
+                var mtd = t.GetTypeInfo().GetDeclaredMethod("Do");
+
+                yield return Expression.Block(new[] { p }, Expression.Assign(p, Expression.New(p.Type.GetTypeInfo().DeclaredConstructors.Single(), Expression.New(t))), Expression.Call(o, mtd), m);
+            }
+
+            // OK - Surprising compared to previous
+            foreach (var t in new[] { typeof(C), typeof(S) })
```

Comment:
Correct. I deemed it surprising in the context of the inconsistency between the compiler and interpreter for a write-back against a MemberExpression for a field vs a property. I expected both to be right or wrong for member access assignments, before I saw the big TODO in the interpreter's compilation logic for MethodCallExpression where the write-back for the instance argument was done always rather than by deep inspection of the assignment target node (and member) kind.